com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -58,47 +89,68 @@ internal class BarracudaModelParamLoader
                 return failedModelChecks;
             }
 
-            var modelApiVersion = (int)model.GetTensorByName(TensorNames.VersionNumber)[0];
+            var modelApiVersion = model.GetVersion();
```

Comment:
Can this actually return -1? Doesn't look like it from https://github.com/Unity-Technologies/ml-agents/pull/5080/files#diff-bf1591d220fea33b1512200891dc1d8000f844664308187d2beaaaff8feb657aR49-R52 (unless that's the actual value).
Seems like this could be combined with the one below.